sweepbatcher/greedy_batch_selection.go

```diff
@@ -44,43 +48,48 @@ func (b *Batcher) greedyAddSweep(ctx context.Context, sweep *sweep) error {
 		batches = append(batches, batchFeeDetails)
 	}
 
-	// Run the algorithm. Get batchId of the best batch, or newBatchSignal
-	// if the best option is to create new batch.
-	batchId, err := selectBatch(
+	// Run the algorithm. Get batchId of possible batches, sorted from best
+	// to worst.
+	batchesIds, err := selectBatches(
 		batches, sweepFeeDetails, newBatchFeeDetails,
 	)
 	if err != nil {
 		return fmt.Errorf("batch selection algorithm failed for sweep "+
 			"%x: %w", sweep.swapHash[:6], err)
 	}
 
-	// If the best option is to start new batch, do it.
-	if batchId == newBatchSignal {
-		return b.spinUpNewBatch(ctx, sweep)
-	}
+	// Try batches, starting with the best.
+	for _, batchId := range batchesIds {
+		// If the best option is to start new batch, do it.
+		if batchId == newBatchSignal {
+			return b.spinUpNewBatch(ctx, sweep)
+		}
 
-	// Locate the batch to add the sweep to.
-	bestBatch, has := b.batches[batchId]
-	if !has {
-		return fmt.Errorf("batch selection algorithm returned "+
-			"batch id %d which doesn't exist, for sweep %x",
-			batchId, sweep.swapHash[:6])
-	}
+		// Locate the batch to add the sweep to.
+		bestBatch, has := b.batches[batchId]
+		if !has {
+			return fmt.Errorf("batch selection algorithm returned "+
+				"batch id %d which doesn't exist, for sweep %x",
+				batchId, sweep.swapHash[:6])
+		}
 
-	// Add the sweep to the batch.
-	accepted, err := bestBatch.addSweep(ctx, sweep)
-	if err != nil {
-		return fmt.Errorf("batch selection algorithm returned "+
-			"batch id %d for sweep %x, but adding failed: %w",
-			batchId, sweep.swapHash[:6], err)
-	}
-	if !accepted {
-		return fmt.Errorf("batch selection algorithm returned "+
-			"batch id %d for sweep %x, but acceptance failed",
-			batchId, sweep.swapHash[:6])
+		// Add the sweep to the batch.
+		accepted, err := bestBatch.addSweep(ctx, sweep)
+		if err != nil {
+			return fmt.Errorf("batch selection algorithm returned "+
+				"batch id %d for sweep %x, but adding failed: "+
+				"%w", batchId, sweep.swapHash[:6], err)
+		}
+		if accepted {
+			return nil
+		}
+
+		log.Infof("Batch selection algorithm returned batch id %d for "+
```

Comment:
nit: I think it's better to do `Debugf` here

Reply:
Done :+1: